### ISSUE TITLE
Doesn't work with Zip64 files

### ISSUE THREAD

**Issue by EbrahimMajdey** [dotnet/docs#7039]
The ZipArchive can't read any entry that is larger than 4,294,967,295 bytes .. which is the old limit for zip files although the OS can read this.. would appreciate if such capabilities is enabled instead of using other packages

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3a46d025-34f1-9ac4-6d29-a8e527cc5f91
* Version Independent ID: 5642d6cc-6abc-e662-bc03-d284ca9cb438
* Content: [ZipArchive Class (System.IO.Compression)](https://docs.microsoft.com/en-us/dotnet/api/system.io.compression.ziparchive?redirectedfrom=MSDN&view=netframework-4.7.2)
* Content Source: [xml/System.IO.Compression/ZipArchive.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System.IO.Compression/ZipArchive.xml)
* Product: **dotnet-api**
* GitHub Login: @mairaw
* Microsoft Alias: **mairaw**

**Comment by svick** [comment]
First, this repo is for feedback on the documentation, so I think this issue doesn't belong here, unless you think this is something that should be pointed out in the documentation.

Second, as far as I can tell from https://github.com/dotnet/corefx/issues/9925, .Net Core does support Zip64 files. .Net Core improvements are sometimes ported to .Net Framework, though I'm not sure how likely is that in this case. Though it might still make sense to ask.

If you still want to provide feedback on .Net Framework, the right place is [Developer Community](https://developercommunity.visualstudio.com/spaces/61/index.html). For .Net Core APIs, the right place is [the corefx repo](https://github.com/dotnet/corefx) (but I don't think this is a .Net Core issue).

**Comment by mairaw** [comment]
@ianhays can you help with this question? Is there something we should add to the docs? Thanks!

**Comment by EbrahimMajdey** [comment]
Yes it should state in the documentation that it doesn't support Zip64, Nowadays it is common to find such large files, would be helpful also if it stays what future frameworks (if any) would support Zip64 .. It is not simple to switch from regular .Net to .Net core in exiting products

**Comment by ianhays** [comment]
@svick is correct. 

There is also a distinction between zip64 and deflate64 and neither of them are exclusive with size limitations:

- Zip64 is a modification of a zip itself and is supported on netcore and netfx.
- Deflate64 is a separate compression algorithm used by some compressors to attempt to more efficiently encode zip entries that are greater than 4gb. Note that I said "entries", not zips. So a 5gb zip with 100 entries usually won't have Deflate64 entries. Again though, this is entirely up to the encoder. The most common source of deflate64-encoded entries in a zip is the Windows explorer zipper.
- Both netfx and netcoreapp support zips larger than 4gb when they're encoded in supported methods and *decoded in Read Mode*.

Since we do support entries larger than 4gb, we shouldn't make any doc changes. We should probably backport Deflate64 support as a functional improvement, but that's a separate issue that should be filed elsewhere as @svick mentioned.
